### PR TITLE
Break out coroutine support to a separate module

### PR DESCRIPTION
**Pull request by evanchooly**
This moves the kotlin related items to a separate module

### CODE REVIEW

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]
What happens if a user adds a co-routine to and endpoint and has not added this dependency? Won't the build fail with a cryptic error message (probably NoClassDefFoundError)?

IMHO if we are going to have a dependency we add then it should be a proper extension, so the deployment code is also in the kotlin module, and in the main RR module we have a little bit of code that can detect a co-routine endpoint and if the kotlin extension is not present print out a nice error message.

**Comment by geoand** [comment]
+1 to what Stuart said.

We should try as hard as we can to provide good and actionable error messages

**Comment by FroMage** [comment]
My impression was that the resteasy-reactive module would depend on the new rr-kotlin module unconditionnally, though this one would keep its kotlin stdlib deps optional, so the current behaviour would not change, but only the ability for product to remove this module, and for us to be able to load it in our IDEs.

**Comment by FroMage** [comment]
I mean, I didn't think the user would have to add any extra explicit dependency for this to work.

**Comment by stuartwdouglas** [comment]
It's listed as an optional dep, so at the moment they do.

**Comment by FroMage** [comment]
I meant that I thought it would not be optional.

**Comment by evanchooly** [comment]
my understanding of the goal was to make sure that those who don't want the kotlin plugin loaded would be able to edit the main resteasy-reactive module.  Productization can use the module as it stands now.  They would just not include the kotlin dep and things should just work.  But this change is primarily driven by development encumbrances.

**Comment by FroMage** [comment]
So just don't make the dependency on `rr-kotlin` optional and make the kotlin runtime optional like it used to be and we'll all be good, no?

**Comment by FroMage** [comment]
Let's keep in mind that once this is merged, https://github.com/quarkusio/quarkus/issues/16739 should be notified and move on.

**Comment by evanchooly** [comment]
> So just don't make the dependency on `rr-kotlin` optional and make the kotlin runtime optional like it used to be and we'll all be good, no?

We could do that.  It's kind of 6 of one and half a dozen of the others.  But I do like that it hides that module from users.  I'll play with while I'm adding the CNFE check for Stuart.

**Comment by evanchooly** [comment]
Actually, that would eliminate the need for the CNFE altogether, really.  +1.

**Comment by evanchooly** [comment]
rearranged the deps as discussed.

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/pom.xml
Comment (FroMage) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/pom.xml:32]: This one used to be optional where you took this from, any reason why it changed?

**Review by FroMage — DISMISSED** [review]

**Comment by FroMage** [comment]
This new set-up looks fine to me, besides the question. But I'd rather check that the others are OK as well, in case I misunderstood our options :)

File: extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/pom.xml
Comment (evanchooly) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive/kotlin/pom.xml]: that was a miscut.  it should be completely gone and now is.

**Review by stuartwdouglas — APPROVED** [review]

**Comment by evanchooly** [comment]
@FroMage looks like merging is blocked by your review.

**Pull request merged**
